JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -21,37 +27,92 @@
     @PersistenceContext
     EntityManager em;
 
-    public ProjectBean() {}
-
-    public String createProject(String projectName, int groupId) {//groupName??
-
-        Group group;
-
-
-        Project project = new Project();
-        Hash hashes = new Hash();
-
-        //get group, corresponding to this id
-        group = getGroup(groupId);
-
-        //-----------------------------------------set information related to project
-        project.setProjectName(projectName);
-        project.setGroup(group);
-
-        em.getTransaction().begin();
-        em.persist(project);
-        em.getTransaction().commit();
+    @EJB
+    UserBean userBean;
 
-        project = getProject(projectName, groupId);
-
-        //------------------------------------------set information related to hashes
-        //  String projectHash = getHash(projectName);
-        hashes.setProject(project);
-        // hashes.setHash(projectHash);
+    public ProjectBean() {}
 
-        em.getTransaction().begin();
-        em.persist(hashes);
-        em.getTransaction().commit();
+    public String createProject(String creator, String hash, String projectName) {//groupName??
```

Comment:
The method is too big and does too much. Need to split it into several methods (extract code that can be considered as some atomic logical operation into a separate private method)